studio/components/interfaces/SQLEditor/SQLEditor.tsx

```diff
@@ -171,8 +173,33 @@ const SQLEditor = () => {
       // Refetching instead of invalidating since invalidate doesn't work with `enabled` flag
       refetchEntityDefinitions()
     },
-    onError(error) {
-      if (id) snap.addResultError(id, error)
+    onError(error: any) {
+      if (id) {
+        if (error.position && monacoRef.current) {
+          const editor = editorRef.current
+          const monaco = monacoRef.current
+
+          const formattedError = error.formattedError ?? ''
+          const lineError = formattedError.slice(formattedError.indexOf('LINE'))
+          const line = Number(lineError.slice(0, lineError.indexOf(':')).split(' ')[1])
+
+          const decorations = editor?.deltaDecorations(
```

Comment:
Is it worth checking `line` is not `NaN` before calling `deltaDecorations()`?

Reply:
oh yeah good point!